### PR TITLE
Refactor OpenAI client initialization and logging in llm_pipeline.py

### PR DESCRIPTION
**Pull request by leonvanbokhorst**
This pull request refactors the OpenAI client initialization and logging in the `llm_pipeline.py` file. The `initialize_openai_client` function has been moved inside the `LLM` class and renamed to `initialize_openai_client`. The logger has also been moved inside the `LLM` class and is now initialized using the `setup_logging` function. Additionally, the `get_response` method has been updated to use the `self.client` and `self.logger` attributes. This refactoring improves the organization and readability of the code.

## Summary by Sourcery

Refactor the OpenAI client initialization and logging by encapsulating them within the LLM class, and update the get_response method to utilize class attributes. Add comprehensive tests for the LLM class to ensure functionality across various scenarios.

Enhancements:
- Refactor the OpenAI client initialization and logging by moving them inside the LLM class, improving code organization and readability.

Tests:
- Add a new test suite for the LLM class in tests/test_llm_pipeline.py, including tests for happy paths, edge cases, and error handling.

### CODE REVIEW

**Review by leonvanbokhorst — COMMENTED** [review]
@sourcery-ai review

File: tests/test_llm_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    # Act
+    response = llm.get_response(
+        user_input, assistant_emotion, energy_level, user_emotion
+    )
+
+    # Assert
+    assert response == expected_response
+    mock_logger.info.assert_called()
+    mock_logger.debug.assert_called()
```
[the remark this answers is not in the document]
Comment (leonvanbokhorst) [reply tests/test_llm_pipeline.py:49]: Sounds good

[the remark this answers is not in the document]
Comment (leonvanbokhorst) [reply tests/test_llm_pipeline.py:49]: Ok fine

**Review by leonvanbokhorst — COMMENTED** [review]
Great stuff

**Review by leonvanbokhorst — COMMENTED** [review]
Great stuff

File: src/llm_pipeline.py [hunk 2]
```diff
@@ -1,75 +1,85 @@
...
-    logger.info("Loading API key from .env file...")
-    load_dotenv()
-    api_key = os.getenv("OPENAI_API_KEY")
-    if not api_key:
-        error_message = "API key not found. Please make sure OPENAI_API_KEY is set in your environment."
-        logger.error(error_message)
-        raise ValueError(error_message)
-    logger.info("API key loaded successfully.")
-    return OpenAI()
+class LLM:
```
[the remark this answers is not in the document]
Comment (leonvanbokhorst) [reply src/llm_pipeline.py:7]: Not now

File: src/llm_pipeline.py [hunk 3]
```diff
@@ -1,75 +1,85 @@
...
+
+        try:
+            self.logger.info("Sending request to OpenAI API...")
+            response = self.client.chat.completions.create(
+                model="gpt-4o-mini", messages=messages
+            )
+            self.logger.info("Response received from OpenAI API.")
+            response_content = response.choices[0].message.content.strip()
+            self.logger.debug(f"Generated response: '{response_content}'")
+            return response_content
```
[the remark this answers is not in the document]
Comment (leonvanbokhorst) [reply src/llm_pipeline.py:69]: maybe later

File: src/llm_pipeline.py [hunk 4]
```diff
@@ -1,75 +1,85 @@
...
+
+            Your response should be empathetic, adaptive, and reflect your current energy level. If your energy is low, your response should show signs of fatigue (e.g., slight delays, hesitations, more casual language). Keep the conversation engaging, yet make sure your own energy level and emotional state are subtly expressed.
+            """,
+            },
+        ]
+
+        try:
+            self.logger.info("Sending request to OpenAI API...")
+            response = self.client.chat.completions.create(
+                model="gpt-4o-mini", messages=messages
```
[the remark this answers is not in the document]
Comment (leonvanbokhorst) [reply src/llm_pipeline.py:64]: I like this. How?
